protocol/binary_test.go

```diff
@@ -376,6 +379,22 @@ func TestStruct(t *testing.T) {
 	checkEncodeDecode(t, wire.TStruct, tests)
 }
 
+func TestStructBeginAndEndEncode(t *testing.T) {
+	var streamBuff bytes.Buffer
+	var err error
```

Comment:
nit: please group these:
```suggestion
	var (
		streamBuff bytes.Buffer
		err error
	)
```

(https://github.com/uber-go/guide/blob/master/style.md#group-similar-declarations)